File: Frontend/src/js/model/employee.js
import $formatter from "../common/formater";

export class Employee {
  address;
  dateOfBirth;
  departmentCode;
  departmentId;
  departmentName;
  email;
  employeeCode;
  employeeId;
  employeeFullName;
  gender;
  identityDate;
  identityNumber;
  identityPlace;
  phoneNumber;
  landlineNumber;
  positionName;
  positionId;
  positionCode;
  bankAccount;
  bankName;
  bankBranch;

  /**
   * Hàm khởi tạo cho Employee
   *
   * Author: Dũng (11/05/2023)
   */
  constructor(e) {
    this.address = e.address ?? "";
    this.dateOfBirth = $formatter.changeFormat(e.dateOfBirth);
    this.departmentCode = e.departmentCode ?? "";
    this.departmentId = e.departmentId ?? "";
    this.departmentName = e.departmentName ?? "";
    this.email = e.email ?? "";
    this.employeeCode = e.employeeCode ?? "";
    this.employeeId = e.employeeId ?? "";
    this.employeeFullName = e.employeeFullName ?? "";
    this.gender = e.gender ?? -1;
    this.identityDate = $formatter.changeFormat(e.identityDate);
    this.identityNumber = e.identityNumber ?? "";
    this.identityPlace = e.identityPlace ?? "";
    this.phoneNumber = e.phoneNumber ?? "";
    this.landlineNumber = e.landlineNumber ?? "";
    this.positionName = e.positionName ?? "";
    this.positionId = e.positionId ?? "";
    this.positionCode = e.positionCode ?? "";
    this.bankAccount = e.bankAccount ?? "";
    this.bankName = e.bankName ?? "";
    this.bankBranch = e.bankBranch ?? "";
  }

  /**
   * Copy dữ liệu từ employee khác vào object hiện tại
   *
   * Author: Dũng (04/06/2023)
   */
  cloneFromOtherEmployee(e) {
    this.address = e.address ?? "";
    this.dateOfBirth = e.dateOfBirth ?? "";
    this.departmentCode = e.departmentCode ?? "";
    this.departmentId = e.departmentId ?? "";
    this.departmentName = e.departmentName ?? "";
    this.email = e.email ?? "";
    this.employeeCode = e.employeeCode ?? "";
    this.employeeId = e.employeeId ?? "";
    this.employeeFullName = e.employeeFullName ?? "";
    this.gender = e.gender ?? -1;
    this.identityDate = e.identityDate ?? "";
    this.identityNumber = e.identityNumber ?? "";
    this.identityPlace = e.identityPlace ?? "";
    this.phoneNumber = e.phoneNumber ?? "";
    this.landlineNumber = e.landlineNumber ?? "";
    this.positionName = e.positionName ?? "";
    this.positionId = e.positionId ?? "";
    this.positionCode = e.positionCode ?? "";
    this.bankAccount = e.bankAccount ?? "";
    this.bankName = e.bankName ?? "";
    this.bankBranch = e.bankBranch ?? "";
  }

  /**
   * Trả về Object với định dạng là Body của request
   * @param {Boolean} isEditApi là api sửa thông tin nhân viên hay không
   *
   * Author: Dũng (11/05/2023)
   */
  convertToApiFormat() {
    let obj = {
      employeeCode: this.employeeCode,
      employeeFullName: this.employeeFullName,
      departmentId: this.departmentId,
      departmentName: this.departmentName,
      positionName: this.positionName,
      dateOfBirth: $formatter.formatDateToApiDate(this.dateOfBirth),
      gender: this.gender != -1 ? this.gender : null,
      identityNumber: this.identityNumber,
      identityDate: $formatter.formatDateToApiDate(this.identityDate),
      identityPlace: this.identityPlace,
      address: this.address,
      bankAccount: this.bankAccount,
      bankBranch: this.bankBranch,
      bankName: this.bankName,
      phoneNumber: this.phoneNumber,
      landlineNumber: this.landlineNumber,
      email: this.email,
    };
    return obj;
  }
}

File: Backend/MISA.WebFresher032023.Demo.Repository/Entities/Output/Employee.cs
using MISA.WebFresher032023.Demo.Common.Enum;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MISA.WebFresher032023.Demo.DataLayer.Entities.Output
{
    // Employee Output Entity
    public class Employee : BaseOutputEntity
    {
        // ID nhân viên
        public Guid EmployeeId { get; set; }
        // Mã nhân viên
        public string EmployeeCode { get; set; }
        // Tên nhân viên
        public string EmployeeFullName { get; set; }
        // ID đơn vị
        public Guid DepartmentId { get; set; }

        // Mã đơn vị
        public string DepartmentCode { get; set; }

        // Tên đơn vị
        public string DepartmentName { get; set; }
        // Vị trí
        public string PositionName { get; set; }
        // Ngày sinh
        public DateTime? DateOfBirth { get; set; }
        // Giá trị giới tính
        public Gender? Gender { get; set; }

        // Số CMND
        public string IdentityNumber { get; set; }
        // Ngày cấp CMND
        public DateTime? IdentityDate { get; set; }
        // Nơi cấp CMND
        public string IdentityPlace { get; set; }
        // Địa chỉ
        public string Address { get; set; }
        // Số ĐT
        public string PhoneNumber { get; set; }
        // Số ĐT cố định
        public string LandlineNumber { get; set; }
        // Email
        public string Email { get; set; }
        // Tài khoản ngân hàng
        public string BankAccount { get; set; }
        // Tên ngân hàng
        public string BankName { get; set; }
        // Chi nhánh ngân hàng
        public string BankBranch { get; set; }
    }
}
